### PR TITLE
feat(banner): update API on main to match next

### PR DESCRIPTION
**Pull request by dierat**
#### High level summary
NOTE: We're planning to merge these API update PRs in a sequence. The sequence will go:
- for each PR, merge in the EDS API update PR
- publish a new package version
- update the package in traject in a PR that has already been reviewed and is waiting to update the component API based on the EDS PR and merge that in
- move on to the next component

To prepare for migrating the `main` branch to use the v1 component versions on `next`, I'm updating the `Banner` component to match the one on `next`. For this series of changes, I'm trying to just change the API, but the `Banner` had some important internal logic changes that affected the API. So I ended up instead just dumping in the whole component from `next` and removed/replaced the parts that relied on tokens or packages we don't have on `main`. (The two main parts that are different from `main` is the spec file and the variant color styles. For the other style tokens, I just used tailwind or standard CSS.)

After landing this, I'll publish a new npm package version, update the EDS package in `traject` and update the component usage to match these changes. That way when we merge the full `next` branch in and deploy, we won't have to update the banners in `traject` and all the other components at the same time. (Several other components are also being updated separately like this).

#### API changes for this component:
- `color` => `variant`
- "alert" `variant` => "error"
- `elevation  === 0` => `isFlat`
- `textContent` has been split into `title` and `description` (and the `BannerTitle` and `BannerMessage` components have been removed; you can just pass in the strings)
- you can also pass `titleAs` and `descriptionAs` props in to change the element they render as
- 1 rem bottom margin added to banner if `isFlat` is falesy to account for the drop shadow
- need to pass in `dismissable` prop to get the Banner close button to show; `onDismiss` is now optional

### Test Plan:
- no type errors
- no big chromatic changes for actual components (just story name changes and there's a slightly wider gap between the heading and the description)

### CODE REVIEW

File: src/components/Banner/Banner.module.css [hunk 1]
```diff
@@ -1,73 +1,137 @@
-.bannerDialog {
-  @apply w-full p-8 rounded border shadow-1;
-  @apply relative;
-  @apply grid gap-4;
-
+/* stylelint-disable comment-whitespace-inside, selector-nested-pattern, declaration-empty-line-before */
```
Comment (dierat) [inline src/components/Banner/Banner.module.css:1]: We seem to have style linting on `main` that conflicts with how we're writing CSS on `next`. 🤷

Comment (anniehu4) [reply src/components/Banner/Banner.module.css:1]: ahh shoot, yeah I think we're using `config-standard` on main but `config-recommended` on next 🤔  i'll make a note to update `main`

File: src/components/Banner/Banner.tsx [hunk 2]
```diff
@@ -60,120 +110,105 @@ export type BannerProps = {
  * ```tsx
  * <Banner
  *   onDismiss={handleDismiss}
- *   textContent={
+ *   text={
  *     <>
- *       <Banner.Title>{bannerTitle}</Banner.Title>
- *       <Banner.Message>{bannerMessage}</Banner.Message>
+ *       <BannerTitle>Some title</Banner.Title>
+ *       <BannerDescription>Some description</Banner.Description>
```
Comment (anniehu4) [inline src/components/Banner/Banner.tsx:116]: should we update the example?

```
<Banner
    onDismiss={handleDismiss}
    title="Some title"
    description="Some description"
/>
```

Comment (dierat) [reply src/components/Banner/Banner.tsx:116]: Yes, thank you! Also updated on `next`: https://github.com/chanzuckerberg/edu-design-system/pull/1021

**Review by anniehu4 — APPROVED** [review]

**Comment by dierat** [comment]
I removed the new dismiss logic from this PR (it was removed from `next` in https://github.com/chanzuckerberg/edu-design-system/pull/1007).

**Review by anniehu4 — APPROVED** [review]

File: src/components/Banner/Banner.module.css (hunk 1 above)
Comment (anniehu4) [reply src/components/Banner/Banner.module.css:1]: https://github.com/chanzuckerberg/edu-design-system/pull/1028 -- I didn't realize this hadn't merged yet, so you might need to merge new changes again 😅  then these suppressions shouldn't be needed

**Pull request merged**